DOCSP-32150: Update Kotlin Delete Users page (#2949)
mongodb/docs-realm: source/sdk/kotlin/users/delete-users.txt

   :depth: 2
   :class: singlecol

> The Kotlin SDK provides methods to create and delete users in your Atlas App 

Author: Updated

Services App.

Create a User
-------------

For most :ref:`authentication methods <authentication-providers>`, Atlas App Services 
automatically creates a :ref:`user object <user-objects>` the first time 
a user authenticates. The only exception is email/password authentication. 
When you use email/password authentication, you must :ref:`register 
<kotlin-register-email-password-user>` and :ref:`confirm 
<kotlin-confirm-a-new-users-email-address>` a user before the user can 
authenticate to an App Services App.

> .. tip:: Google and Apple Account Deletion Requirements

Author: Added

   `Google <https://support.google.com/googleplay/android-developer/answer/13316080?sjid=9059006274298096173-NA#account_deletion>`__ 
   and :apple:`Apple <app-store/review/guidelines/#5.1.1>` require that 
   applications listed through their respective App Stores must give any 
   user who creates an account the option to delete the account. Whether you 
   use an authentication method where you must manually register a user, 
   such as email/password authentication, or one that that automatically 
   creates a user, such as Sign-In with Apple, you must implement 
   :ref:`user account deletion <delete-user>`.

.. _kotlin-remove-a-user:

Remove a User
-------------

You can call the `remove <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/remove.html>`__

> method on a user to remove the ``User`` object from your Atlas App Services App. This method *does not* delete the ``User`` object from the Atlas App Services server.

Author: updated

The ``user.remove()`` method: 

- :ref:`Logs the user <kotlin-logout>` out if they are logged in.

> - Deletes locally stored data associated with the user from the device.

Reviewer: I wonder if we want to be more specific here and say "synced realms?" I think this could be ambiguous otherwise.
```suggestion
- Deletes synced realms associated with the user from the device.
```

Author: updated

- Sets the
  `User.State <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/-state/index.html>`__ 
  to ``REMOVED``. 

> Because removing a user deletes any local realms owned by the user, you 

Reviewer: Local realms don't really have a concept of being owned by a user.
```suggestion
Because removing a user deletes any synced realms owned by the user, you 
```

Author: updated

should only call this method after closing the user's realms.

.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.remove-user.kt
   :language: kotlin

Delete a User
-------------

You can call the 
`delete <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/delete.html>`__ 
method on a logged-in user to permanently delete the ``User`` object 

> from your Atlas App Services App *and* the App Services server.

Reviewer: I think these things are both saying the same thing. We need to get the client app in, and I think "App Services server" isn't a term we generally use in the docs.
```suggestion
from your client app *and* the Atlas App Services App.
```

Author: updated

The ``user.delete()`` method:

> - Deletes locally stored data associated with the user from the device.

Reviewer: If you take my suggestion above about changing this to synced realms, we should do the same here.
```suggestion
- Deletes synced realms associated with the user from the device.
```

Author: updated

- Deletes the ``User`` object from the Atlas App Services server.
- Sets the
  `User.State <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/-state/index.html>`__ 
  to ``REMOVED``.  

> Because this method deletes any local realms owned by the user, you 

Reviewer: ```suggestion
Because this method deletes any synced realms owned by the user, you 
```

Author: updated

should only call this method after closing the user's realms.

> .. important:: Deleting a User Doesn't Delete User Metadata

Reviewer: Same Q as above - can we make this an `include` and add it to all the SDKs except Swift?

Author: we can if we don't mind the links being updates to more generic app services links (vs. sdk-specific procedures for metadata and custom user data)
